reference-implementation/lib/transform-stream.js

```diff
@@ -354,17 +359,20 @@ module.exports = class TransformStream {
       startPromise_resolve = resolve;
     });
 
-    const sink = new TransformStreamSink(this, startPromise);
-
-    this._writable = new WritableStream(sink, transformer.writableStrategy);
-
     const source = new TransformStreamSource(this, startPromise);
 
     this._readable = new ReadableStream(source, transformer.readableStrategy);
 
+    const sink = new TransformStreamSink(this, startPromise);
+
+    this._writable = new WritableStream(sink, transformer.writableStrategy);
+
     assert(this._writableController !== undefined);
     assert(this._readableController !== undefined);
 
+    const desiredSize = this._readableController.desiredSize;
+    TransformStreamSetBackpressure(this, desiredSize <= 0);
```

Comment:
Are there ever going to be any active reads when we're still inside the TransformStream constructor? I thought it was physically impossible for someone to have called `ts.readable.getReader().read()` at this point.

And it already depends on `pull()` invocation and sets backpressure to false whenever `pull()` is called, so I'm not sure what you're suggesting here.